src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultParameterValidator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class DefaultParameterValidator : IParameterValidator
+    {
+        private readonly ValidatorCache _validatorCache;
+        private readonly IModelValidatorProvider _validatorProvider;
+        private readonly IExtendedModelMetadataProvider _modelMetadataProvider;
+
+        public DefaultParameterValidator(
+            IModelMetadataProvider modelMetadataProvider,
+            IList<IModelValidatorProvider> validatorProviders)
+        {
+            if (modelMetadataProvider == null)
+            {
+                throw new ArgumentNullException(nameof(modelMetadataProvider));
+            }
+
+            if (validatorProviders == null)
+            {
+                throw new ArgumentNullException(nameof(validatorProviders));
+            }
+
+            _validatorCache = new ValidatorCache();
+            _validatorProvider = new CompositeModelValidatorProvider(validatorProviders);
+
+            // Only the extended metadata provider interface provides information about properties,
+            // so if we don't get that type, this validator will do nothing.
+            _modelMetadataProvider = modelMetadataProvider as IExtendedModelMetadataProvider;
+        }
+
+        public void Validate(ActionContext actionContext, ParameterDescriptor parameterDescriptor, ModelMetadata modelMetadata, bool isValueSet, object parameterValue, ModelBindingContext bindingContext)
```

Comment:
Holy signature batman!

Reply:
This is why I'm suggesting merging this with the binding responsibilities. A bunch of details from model binding are leaking in here.

Reply:
OK done.